docs: add documentation for threading tuning parameters

sagarth opened the pull request:
## Pull Request Description
This PR adds documentation for tuning parameters related to multi-threading (MPI_THREAD_MULTIPLE).

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [ ] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

hzhou commented:
👍 Looks good. Ping me when it is ready to review.

hzhou commented on doc/mpich/TUNING_PARAMETERS:
Could you wrap the text to width of less than 80 chars?

sagarth replied:
Done

hzhou commented on doc/mpich/TUNING_PARAMETERS:
Need explain how to configure/select between "global" and "find-grained".

sagarth replied:
I'll add the details.

sagarth replied:
Done

hzhou commented on doc/mpich/TUNING_PARAMETERS:
Document the configure option and runtime cvars.

sagarth replied:
I'll add them.

sagarth replied:
Done

hzhou reviewed (changes requested):
It's good explanations. However, it is immediately clear for a user how to configure/tune.  List configure options and cvars in separate code snippet block should help.

sagarth commented:
@hzhou I've added a fixup patch with your requested changes. Could you check if it looks fine? I'll squash it after the review.

hzhou commented on doc/mpich/TUNING_PARAMETERS:
```diff
@@ -43,12 +43,26 @@ VCI lock and a `lockless` mode where MPI does not grab a lock and relies on the
+A threading synchronization strategy can be selected at the configuration time
+as:
+  --enable-thread-cs=global to select coarse-grained locking, and
+  --enable-thread-cs=per-vci to select fine-grained locking.
```
Can you break the line so the command snippet is on its own line?

sagarth replied:
I was debating with myself about it. Sure, let me do that.

sagarth replied:
Done

hzhou commented on doc/mpich/TUNING_PARAMETERS:
```diff
@@ -0,0 +1,117 @@
+https://wiki.mpich.org/mpich/index.php/Using_the_Hydra_Process_Manager#Process-core_Binding.
+
+
+
```
Extra empty lines.

hzhou commented on doc/mpich/TUNING_PARAMETERS:
```diff
@@ -0,0 +1,117 @@
+
+**1. Multi-threading**
+
+
```
extra empty line

hzhou reviewed (approved):
I notice you are using MARKDOWN formating throughout. How about rename this document into `tuning_parameters.md` and then fix the part that doesn't look right (like headings)?

I am approving it as I am OK to merge it as is. It's up to you to make additional changes. Just let me know when you are happy for it to merge.

sagarth commented:
Fixed the space issues. Also converted the doc file to `.md` and fixed some of the formatting issues. I kept the conversion on a separate patch though.

hzhou commented:
How about add a title?

sagarth commented:
> How about add a title?

Is the how we add title to a MARKDOWN file? : 
`% MPICH Tuning Parameters`

hzhou commented:
> > How about add a title?
> 
> Is the how we add title to a MARKDOWN file? : `% MPICH Tuning Parameters`

Title is just bigger heading. 

# This is a title

## Section title

sagarth commented:
Updated
